### PR TITLE
New message notification for AI assistant 

### PR DESCRIPTION
**Pull request by habdelra**
This PR adds a new message indicator. Basically this means refactoring the read receipt to only trigger after a message has been scrolled into view. As well as to add new features to decouple the auto scrolling from the appearance of new messages.

new_message2

### CODE REVIEW

File: packages/host/app/components/ai-assistant/message/index.gts [hunk 1]
```diff
@@ -31,16 +33,82 @@ interface Signature {
...
+    { index, registerScroller }: MessageScrollerSignature['Args']['Named'],
+  ) {
+    if (!this.hasRegistered) {
+      this.hasRegistered = true;
+      registerScroller({
+        index,
+        element,
+        scrollTo: () => element.scrollIntoView(),
+      });
+      console.log(`registering message ${index}`);
```
Comment (habdelra) [inline packages/host/app/components/ai-assistant/message/index.gts:76]: cleanup

File: packages/host/app/components/matrix/room-message.gts [hunk 2]
```diff
@@ -429,6 +376,8 @@ export default class RoomMessage extends Component<Signature> {
     }
   }
 
+  // TODO need to reevalutate this modifier--do we want to hijack the scroll
+  // when the user views the code?
```
Comment (habdelra) [inline packages/host/app/components/matrix/room-message.gts:380]: TODO

**Review by jurgenwerk — COMMENTED** [review]
I'm seeing glitches where the new message button flickers when the bot responds. See in this video where the button appears and disappears very quickly, twice: 

https://github.com/user-attachments/assets/f049158b-7201-4363-bd08-dbf4e9bd9976

**Comment by habdelra** [comment]
> I'm seeing glitches where the new message button flickers when the bot responds. See in this video where the button appears and disappears very quickly, twice:
> 
>  Screen.Recording.2024-10-25.at.09.40.13.mov

Thanks, I’ll fix that

**Comment by habdelra** [comment]
@jurgenwerk ok, the glitch issue should be fixed in https://github.com/cardstack/boxel/pull/1715/commits/056bb0d3c4677b847fb8a69d7261aa910329c548. wanna check again?

**Review by jurgenwerk — APPROVED** [review]
I tested it again (using pr preview) and now it works great!

**Pull request merged**